contributing.md

```diff
@@ -0,0 +1,92 @@
+      have, make sure to contact them to potentially collaborate.
+   3. If nobody has been assigned for the issue and you would like to work on it,
+      make a comment on the issue to inform the community of your intentions to
+      begin work.
+   4. Follow standard GitHub best practices, i.e. fork the repo, branch from the
+      HEAD of `main`, make commits, and submit a PR to `main`
+      - For core developers working within the repo, to ensure a clear ownership
+        of branches, branches must be named with the convention `{moniker}/{issue#}-branch-name`.
+   5. Be sure to submit the PR in `Draft` mode. Submit your PR early, even if
+      it's incomplete as this indicates to the community you're working on
+      something and allows them to provide comments early in the development
+      process.
+   6. When the code is complete it can be marked `Ready for Review` and follow
+      the PR readiness checklist.
+
+## Coding Guidelines
+
+We follow the Cosmos SDK [Coding Guidelines](https://github.com/cosmos/cosmos-sdk/blob/main/CODING_GUIDELINES.md). Specifically:
+
+- API & Design SHOULD be proposed and reviewed before the main implementaion.
+- Minimize code duplication
+- Define [Acceptance tests](https://github.com/cosmos/cosmos-sdk/blob/main/CODING_GUIDELINES.md#acceptance-tests) or while implementing new features.
+  - Prefer use of acceptance test framework, like [gocuke](https://github.com/regen-network/gocuke/)
+  - For unit tests or integration tests use [go mock](https://github.com/golang/mock) for creating mocks. Generate mock interface implementations using `go generate`.
+
+## Design Documents
+
+When proposing a design decision for the Ojo network, please start by
+opening an [Issue](https://github.com/ojo-network/ojo/issues/new/choose) or a
+[Discussion](https://github.com/ojo-network/ojo/discussions/new) with a summary
+of the proposal.
+
+Once the proposal has been discussed and there is rough alignment on a high-level
+approach to the design, a [design doc](https://github.com/ojo-network/ojo/blob/main/docs/design/README.md) can be drafted in a dedicated pull request.
+We are following this process to ensure all involved parties are in agreement before any party begins coding the proposed implementation.
+
+## Branching Model
+
+The Ojo network repo adheres to the [trunk based development branching](https://trunkbaseddevelopment.com/)
+model and utilizes [semantic versioning](https://semver.org/).
+
+### PR Targeting
+
+Ensure that you base and target your PR against the `main` branch.
+
+All feature additions should be targeted against `main`. Bug fixes for an
+outstanding release candidate should be targeted against the release candidate
+branch.
+
+### PR & Merge Procedure
+
+- Ensure the PR branch is rebased on `main`.
+- Ensure you provided unit tests and integration tests.
+- Run `make test-unit test-e2e` to ensure that all tests pass.
```

Comment:
I don't think the e2e tests exist yet. It is in the make file but I'm betting it doesn't work. There is a lot of other stuff that is still carried over in the ojo Makefile. Do we want to leave this because we'll add it soon, or update it as we go?

Reply:
let's update it as we go, I'll remove this stuff 😉

Reply:
will remove the e2e stuff in #39 since I touch these files anyways

Reply:
7e1ff6d8930d9b90bea40c1b532f4b446021a75d & 201029a953077162050f75ea20cea4c7267c9f72